modules/ggipc-client/src/client.c

```diff
@@ -109,14 +108,6 @@ static GglError get_message(
         }
     }
 
-    if (payload != NULL) {
-        ret = ggl_json_decode_destructive(msg->payload, alloc, payload);
-        if (ret != GGL_ERR_OK) {
-            return ret;
-        }
-        msg->payload.len = 0;
-    }
-
```

Comment:
Why not decode in the `get_message` call? I realize bubbling this up to the `ggipc_call` reduces the number of parameters but decoding the message seems like it should be part of getting the message.

Reply:
Also error payloads are not parsed as JSON currently, but used as a string.